core/src/main/scala/com/criteo/cuttle/Executor.scala

```diff
@@ -827,16 +847,29 @@ class Executor[S <: Scheduling] private[cuttle] (val platforms: Seq[ExecutionPla
                 .toLeft {
                   val nextExecutionId = utils.randomUUID
 
+                  val streams = new ExecutionStreams {
+                    def writeln(str: CharSequence) = ExecutionStreams.writeln(nextExecutionId, str)
+                  }
+
+                  // wrap the execution context so that we can register the name of the thread of each
+                  // runnable (and thus future) that will be run by the side effect.
+                  val sideEffectExecutionContext = SideEffectExecutionContext.wrap(runnable => new Runnable {
+                      override def run(): Unit = {
+                        val tName = Thread.currentThread().getName
+                        Executor.threadNamesToStreams.put(tName, streams)
```

Comment:
Thread names are not guaranteed to be unique. Why not put a reference to the `Thread` instead?

Reply:
Only the thread name is exposed in the log4j LoggingEvent